lib/evm/interpreter.js

```diff
@@ -0,0 +1,206 @@
+const promisify = require('util.promisify')
+const BN = require('bn.js')
+const ethUtil = require('ethereumjs-util')
+const { ERROR } = require('../exceptions')
+const Loop = require('./loop')
+const { StorageReader } = require('../state')
+const PStateManager = require('../state/promisified')
+
+module.exports = class Interpreter {
+  constructor (vm, txContext, block, storageReader) {
```

Comment:
General note: I think every access through _vm here is an indicator that we have some misplacement of code or suboptimal structure.